program/src/distribute_ncn_vault_rewards.rs

```diff
@@ -0,0 +1,123 @@
+use jito_bytemuck::AccountDeserialize;
+use jito_restaking_core::{config::Config, ncn::Ncn, operator::Operator};
+use jito_tip_router_core::{
+    error::TipRouterError, loaders::load_ncn_epoch, ncn_config::NcnConfig,
+    ncn_fee_group::NcnFeeGroup, ncn_reward_router::NcnRewardRouter,
+};
+use jito_vault_core::vault::Vault;
+use solana_program::{
+    account_info::AccountInfo, clock::Clock, entrypoint::ProgramResult, msg,
+    program_error::ProgramError, pubkey::Pubkey, rent::Rent, sysvar::Sysvar,
+};
+
+/// Can be backfilled for previous epochs
+pub fn process_distribute_ncn_vault_rewards(
+    program_id: &Pubkey,
+    accounts: &[AccountInfo],
+    ncn_fee_group: u8,
+    first_slot_of_ncn_epoch: Option<u64>,
+) -> ProgramResult {
+    let [restaking_config, ncn_config, ncn, operator, vault, ncn_reward_router, restaking_program, vault_program] =
+        accounts
+    else {
+        return Err(ProgramError::NotEnoughAccountKeys);
+    };
+
+    if restaking_program.key.ne(&jito_restaking_program::id()) {
+        msg!("Incorrect restaking program ID");
+        return Err(ProgramError::InvalidAccountData);
+    }
+
+    if vault_program.key.ne(&jito_vault_program::id()) {
+        msg!("Incorrect vault program ID");
+        return Err(ProgramError::InvalidAccountData);
+    }
```

Comment:
curious if there's purpose to passing in the vault program as account if we're gonna require it to be a certain value and don't need to invoke the program?

Reply:
vault_program is needed to call `Vault::load`

Reply:
sure but if the instruction fails if we pass in anything other than vault::id() why not just pass vault::id() into Vault::load and save an account from the instruction? (same question for the restaking program checks, but prob getting rid of those when we factor out ncn_epoch)

Reply:
good point